app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,78 @@
+class TweetsController < ApplicationController
+
+
+  get '/tweets' do
+    if logged_in?
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if session[:user_id]
+      @user = current_user
+      erb :'tweets/new_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/edit_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/delete' do
+
+  end
+
+  post '/tweets/new' do
```

Comment:
Typically, tweet creation would be posted to `/tweets` instead of `/tweets/new`.